packages/fxa-settings/src/pages/ResetPassword/index.test.tsx

```diff
@@ -0,0 +1,167 @@
+// import { logViewEvent } from '../../lib/metrics';
+import ResetPassword from '.';
+import { MOCK_SERVICE_NAME, MOCK_EMAIL } from './mocks';
+
+jest.mock('../../lib/metrics', () => ({
+  usePageViewEvent: jest.fn(),
+  logViewEvent: jest.fn(),
+  settingsViewName: 'settings',
+}));
+
+const mockNavigate = jest.fn();
+jest.mock('@reach/router', () => ({
+  ...jest.requireActual('@reach/router'),
+  useNavigate: () => mockNavigate,
+}));
+
+describe('PageResetPassword', () => {
+  // TODO: enable l10n tests when they've been updated to handle embedded tags in ftl strings
+  // let bundle: FluentBundle;
+  // beforeAll(async () => {
+  //   bundle = await getFtlBundle('settings');
+  // });
+
+  it('renders as expected when no props are provided', () => {
+    render(<ResetPassword />);
+    // testAllL10n(screen, bundle);
+
+    const headingEl = screen.getByRole('heading', { level: 1 });
+    expect(headingEl).toHaveTextContent(
+      'Reset password to continue to account settings'
+    );
+    expect(screen.getByTestId('warning-message-container')).toBeInTheDocument();
+    expect(screen.getByLabelText('Email')).toBeInTheDocument();
+    expect(
+      screen.getByRole('button', { name: 'Begin Reset' })
+    ).toBeInTheDocument();
+    // when forceEmail is NOT provided as a prop, the optional read-only email should not be rendered
+    const forcedEmailEl = screen.queryByTestId('reset-password-force-email');
+    expect(forcedEmailEl).not.toBeInTheDocument();
+    // when 'canGoBack: false' or not passed as prop, the optional RememberPassword link component should not be rendered
+    expect(screen.queryByRole('link')).not.toBeInTheDocument();
+  });
+
+  it('renders a custom service name in the header when it is provided', () => {
+    render(<ResetPassword serviceName={MOCK_SERVICE_NAME} />);
+    const headingEl = screen.getByRole('heading', { level: 1 });
+    expect(headingEl).toHaveTextContent(
+      `Reset password to continue to ${MOCK_SERVICE_NAME}`
+    );
+  });
+
+  it('renders a read-only email but no text input when forceEmail is provided as a prop', () => {
+    render(<ResetPassword forceEmail={MOCK_EMAIL} />);
+    expect(
+      screen.getByTestId('reset-password-force-email')
+    ).toBeInTheDocument();
+    expect(screen.queryByLabelText('Email')).not.toBeInTheDocument();
+  });
+
+  it('renders a "Remember your password?" link if "canGoBack: true"', () => {
+    render(<ResetPassword canGoBack={true} />);
+    expect(
+      screen.getByRole('link', { name: 'Remember your password? Sign in' })
+    ).toBeInTheDocument();
+  });
+
+  it('emits a metrics event on render', () => {
+    render(<ResetPassword />);
+    expect(usePageViewEvent).toHaveBeenCalledWith(`reset-password`, {
+      entrypoint_variation: 'react',
+    });
+  });
+
+  // TODO activate testing for success when account.resetPassword created
+  // currently, works if index.tsx line 87 commented out (await account...)
```

Comment:
I think it might be worth pulling this stuff out and pushing it to a separate branch that we can link in the functional ticket. Great to have it started though.